Fix email user

jamiejamison opened the pull request:
This is a dumb fix but it works. If we create an environment variable in Jenkins named "LC_BUILD_USER_ID" and set it to BUILD_USER_ID, the name of the user running the build, we can inject it into the build process and ssh will allow it to pass through because it looks like a locale environment variable.

jamiejamison committed:
Fix for e-mail problem

jamiejamison committed:
Bump version

gorism commented:
Just so I understand the problem:

1. We're now kicking off the deploys from Jenkins, but the actual execution is done from a bastion server
2. Jenkins uses ssh to remotely kick off the build
3. On the bastion server, it's a shared login executing the build.  The context of the user who kicked off the build isn't available
4. The user that kicked off the build is captured on the Jenkins side in an environment variable, but SSH isn't including that in the environment variables on the bastion server
5. Using an environment variable that looks like a locale will get SSH to include that environment variable for the user session on the bastion server.

Did I get that correct?

gorism reviewed (commented):
Just so I understand the problem:

1. We're now kicking off the deploys from Jenkins, but the actual execution is done from a bastion server
2. Jenkins uses ssh to remotely kick off the build
3. On the bastion server, it's a shared login executing the build. The context of the user who kicked off the build isn't available
4. The user that kicked off the build is captured on the Jenkins side in an environment variable, but SSH isn't including that in the environment variables on the bastion server
5. Using an environment variable that looks like a locale will get SSH to include that environment variable for the user session on the bastion server.

Did I get that correct?

jamiejamison committed:
Update presenter.rb

I changed LC_BUILD_USER_ID to BUILD_USER_ID and used ENV.fetch instead of the trinary.

jamiejamison commented:
You got it right but I didn't. After some more experimentation I found that setting environment variables in Jenkins doesn't work exactly like setting them in a bash shell. I figured out another way to get the value of BUILD_USER_ID from the Jenkins server to the deployer and changed the variable name to match what we set up when I modified the infinity deploy script to deal with deploying over ssh.
